src/processing_test.rs

```diff
@@ -0,0 +1,32 @@
+#[cfg(test)]
+
+use processing::Processing;
+#[allow(unused_imports)]
+use error::ErrorType;
+
+#[test]
+fn test_read_from_directory() {
+    let mut processing = Processing::default();
+    match processing.read_from_directory("test/example_md/real_md") {
```

Comment:
here you could use simply `unwrap()` or `assert!(processing.read_from_directory(...).is_ok())`